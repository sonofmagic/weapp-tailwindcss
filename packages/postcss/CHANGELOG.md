# @weapp-tailwindcss/postcss

## 1.0.0-alpha.1

### Major Changes

- [`bafd149`](https://github.com/sonofmagic/weapp-tailwindcss/commit/bafd149f0510b30cbf95711223583055023e7875) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: monorepo changes

### Patch Changes

- Updated dependencies [[`bafd149`](https://github.com/sonofmagic/weapp-tailwindcss/commit/bafd149f0510b30cbf95711223583055023e7875)]:
  - @weapp-tailwindcss/shared@1.0.0-alpha.1

## 0.0.1-alpha.0

### Patch Changes

- Updated dependencies [[`d50d95d`](https://github.com/sonofmagic/weapp-tailwindcss/commit/d50d95d04e1c6b7c3ff32acc0d9894d3c0f06d22)]:
  - @weapp-tailwindcss/shared@0.0.1-alpha.0
