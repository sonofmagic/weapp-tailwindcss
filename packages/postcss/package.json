{
  "name": "@weapp-tailwindcss/postcss",
  "version": "1.0.0-alpha.2",
  "description": "@weapp-tailwindcss/postcss",
  "author": "ice breaker <1324318532@qq.com>",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/sonofmagic/monorepo-template.git",
    "directory": "packages/bar"
  },
  "bugs": {
    "url": "https://github.com/sonofmagic/monorepo-template/issues"
  },
  "keywords": [],
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.mjs",
      "require": "./dist/index.js"
    },
    "./types": {
      "types": "./dist/types.d.ts",
      "import": "./dist/types.mjs",
      "require": "./dist/types.js"
    }
  },
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "dev": "tsup --watch --sourcemap",
    "build": "tsup",
    "test": "vitest run",
    "test:dev": "vitest",
    "release": "pnpm publish",
    "lint": "eslint .",
    "lint:fix": "eslint . --fix"
  },
  "publishConfig": {
    "exports": {
      ".": {
        "types": "./dist/index.d.ts",
        "import": "./dist/index.mjs",
        "require": "./dist/index.js"
      },
      "./types": {
        "types": "./dist/types.d.ts",
        "import": "./dist/types.mjs",
        "require": "./dist/types.js"
      }
    },
    "main": "./dist/index.js",
    "module": "./dist/index.mjs",
    "types": "./dist/index.d.ts"
  },
  "dependencies": {
    "@csstools/postcss-is-pseudo-class": "^5.0.1",
    "@weapp-core/escape": "~3.0.2",
    "@weapp-tailwindcss/shared": "workspace:*",
    "postcss": "~8.4.49",
    "postcss-rem-to-responsive-pixel": "~6.0.2",
    "postcss-selector-parser": "~7.0.0"
  },
  "devDependencies": {
    "@weapp-tailwindcss/mangle": "workspace:*"
  }
}
