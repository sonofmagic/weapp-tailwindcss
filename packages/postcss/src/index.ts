// 统一导出入口，供外部调用端按需引用核心能力
export * from './handler'
export {
  createStylePipeline,
  type PipelineNodeContext,
  type PipelineNodeCursor,
  type PipelineStage,
  type ResolvedPipelineNode,
  type StyleProcessingPipeline,
} from './pipeline'
export { createInjectPreflight } from './preflight'
export { internalCssSelectorReplacer } from './shared'
export * from './types'
