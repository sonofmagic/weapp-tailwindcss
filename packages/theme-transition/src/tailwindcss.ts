import type { Config } from 'tailwindcss'
import type { CSSRuleObject } from 'tailwindcss/types/config'
import defu from 'defu'
import plugin from 'tailwindcss/plugin'

export interface ThemeTransitionPluginOptions {
  zIndex?: {
    ceiling?: string | number
    floor?: string | number
  }
  // darkSelector?: string
  baseCss?: CSSRuleObject
  darkMode?: Config['darkMode']
}

type DeepRequired<T> = Required<{
  [K in keyof T]: T[K] extends Required<T[K]> ? T[K] : DeepRequired<T[K]>
}>

// https://github.com/tailwindlabs/tailwindcss/blob/557ed8ccecccf703296c186b6e996edf9933e1b7/packages/tailwindcss/src/compat/plugin-api.ts#L28
// https://github.com/tailwindlabs/tailwindcss/blob/557ed8ccecccf703296c186b6e996edf9933e1b7/packages/tailwindcss/src/compat/plugin-api.ts#L92
// https://github.com/tailwindlabs/tailwindcss/tree/main
// https://github.com/tailwindlabs/tailwindcss/blob/557ed8ccecccf703296c186b6e996edf9933e1b7/packages/tailwindcss/src/compat/dark-mode.ts#L4
export function themeTransitionPlugin(themeTransitionPluginOptions?: ThemeTransitionPluginOptions) {
  const { baseCss, zIndex, darkMode: specDarkMode } = defu<
    DeepRequired<ThemeTransitionPluginOptions>,
    ThemeTransitionPluginOptions[]
  >(
    themeTransitionPluginOptions,
    {
      baseCss: {
        mixBlendMode: 'normal',
        animation: 'none',
      },
      zIndex: {
        ceiling: 2147483646,
        floor: 1,
      },
    },
  )
  const floor = zIndex.floor.toString()
  const ceiling = zIndex.ceiling.toString()
  return plugin(({ addBase, config }) => {
    const darkMode: Config['darkMode'] = specDarkMode ?? config('darkMode')
    let [mode, selector = '.dark'] = Array.isArray(darkMode) ? darkMode : [darkMode]
    if (mode === 'variant') {
      let formats

      if (Array.isArray(selector)) {
        formats = selector
      }
      else if (typeof selector === 'function') {
        formats = selector
      }
      else if (typeof selector === 'string') {
        formats = [selector]
      }

      if (Array.isArray(formats)) {
        for (const format of formats) {
          if (format === '.dark') {
            mode = undefined
            console.warn(
              'When using `variant` for `darkMode`, you must provide a selector.\nExample: `darkMode: ["variant", ".your-selector &"]`',
            )
          }
          else if (!format.includes('&')) {
            mode = undefined
            console.warn(
              'When using `variant` for `darkMode`, your selector must contain `&`.\nExample `darkMode: ["variant", ".your-selector &"]`',
            )
          }
        }
      }

      selector = formats as any
    }
    const rule: CSSRuleObject = {
      '::view-transition-old(root),\n::view-transition-new(root)': baseCss,
      '::view-transition-old(root)': {
        zIndex: floor,
      },
      '::view-transition-new(root)': {
        zIndex: ceiling,
      },
    }
    if (mode === 'selector' || mode === 'class') {
      rule[`${selector}::view-transition-old(root)`] = {
        zIndex: ceiling,
      }
      rule[`${selector}::view-transition-new(root)`] = {
        zIndex: floor,
      }
    }
    else if (mode === 'media') {
      rule['@media (prefers-color-scheme: dark)'] = {
        '::view-transition-old(root)': {
          zIndex: ceiling,
        },
        '::view-transition-new(root)': {
          zIndex: floor,
        },
      }
    }
    // || mode === 'variant'

    addBase(
      rule,
    )
  })
}

export default themeTransitionPlugin
