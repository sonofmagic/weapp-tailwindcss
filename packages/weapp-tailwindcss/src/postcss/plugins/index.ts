import type { AcceptedPlugin } from 'postcss'
// @ts-expect-error
import postcssIsPseudoClass from '@csstools/postcss-is-pseudo-class'
import postcssRem2rpx from 'postcss-rem-to-responsive-pixel'
import type { IStyleHandlerOptions } from '../../types'
import { postcssWeappTailwindcssPrePlugin } from './pre'
import { postcssWeappTailwindcssPostPlugin } from './post'
import { createContext } from './ctx'

export function getPlugins(options: IStyleHandlerOptions) {
  const ctx = createContext()
  options.ctx = ctx
  const plugins: AcceptedPlugin[] = [
    ...(options.postcssOptions?.plugins ?? []),
    postcssWeappTailwindcssPrePlugin(options),
    postcssIsPseudoClass({
      specificityMatchingName: 'weapp-tw-ig',
    }),
  ]
  if (options.rem2rpx) {
    plugins.push(
      postcssRem2rpx(
        typeof options.rem2rpx === 'object'
          ? options.rem2rpx
          : {
              rootValue: 32,
              propList: ['*'],
              transformUnit: 'rpx',
            },
      ),
    )
  }
  plugins.push(postcssWeappTailwindcssPostPlugin(options))
  return plugins
}

export { postcssWeappTailwindcssPostPlugin } from './post'
export { postcssWeappTailwindcssPrePlugin } from './pre'
// @ts-expect-error
export { default as postcssIsPseudoClass } from '@csstools/postcss-is-pseudo-class'
export { default as postcssRem2rpx } from 'postcss-rem-to-responsive-pixel'
