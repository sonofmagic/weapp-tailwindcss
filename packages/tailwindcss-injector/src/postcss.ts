import type { Input, PluginCreator } from 'postcss'
import type { Config } from 'tailwindcss'
import type { Options } from './types'
import postcss from 'postcss'
import tailwindcss from 'tailwindcss'
import { loadConfig } from 'tailwindcss-config'
import { getConfig } from './config'
import { postcssPlugin } from './constants'
import { regExpTest, removeFileExtension } from './utils'
import { getDepFiles } from './wxml'

const configCache = new Map<string, Promise<Config | undefined>>()

async function loadTailwindConfigOnce(cwd: string, configPath: string) {
  const cacheKey = `${cwd}::${configPath}`
  let cached = configCache.get(cacheKey)
  if (!cached) {
    cached = loadConfig({
      cwd,
      config: configPath,
    }).then(result => result?.config as Config | undefined)
    configCache.set(cacheKey, cached)
  }
  return cached
}

function cloneConfig(config: Config | undefined): Config | undefined {
  if (!config) {
    return config
  }

  return {
    ...config,
  }
}

async function resolveTailwindConfig(config: Options['config'], cwd: string, sourceInput: Input | undefined) {
  if (!config) {
    return undefined
  }

  const resolved = typeof config === 'function' ? config(sourceInput) : config

  if (typeof resolved === 'string') {
    const loaded = await loadTailwindConfigOnce(cwd, resolved)
    return cloneConfig(loaded)
  }

  return cloneConfig(resolved as Config)
}

// function isObject(obj: any): obj is object {
//   return typeof obj === 'object' && obj !== null
// }
export type {
  Options,
}

const creator: PluginCreator<Partial<Options>> = (options) => {
  const { config, filter, directiveParams, cwd, extensions, insertAfterAtRulesNames, insertAfterComments } = getConfig(options)

  const extensionsGlob = `{${extensions.join(',')}}`

  return {
    postcssPlugin,
    plugins: [
      {
        postcssPlugin: `${postcssPlugin}:post`,
        async Once(root, helpers) {
          const sourceInput = root.source?.input
          if (filter(sourceInput)) {
            const nodes = root.nodes ?? []
            let atruleAnchorIndex = -1
            let commentAnchorIndex = -1
            const directivePresence = new Set<string>()

            nodes.forEach((node, index) => {
              if (node.type === 'atrule') {
                const nameMatches = insertAfterAtRulesNames.includes(node.name)
                if (nameMatches) {
                  atruleAnchorIndex = Math.max(atruleAnchorIndex, index)
                }
                if (typeof node.params === 'string') {
                  directivePresence.add(node.params)
                }
              }
              else if (node.type === 'comment' && regExpTest(insertAfterComments, node.text)) {
                commentAnchorIndex = Math.max(commentAnchorIndex, index)
              }
            })

            const anchorIndex = Math.max(atruleAnchorIndex, commentAnchorIndex)
            const anchorNode = anchorIndex >= 0 ? nodes[anchorIndex] : undefined
            let lastInserted = anchorNode

            for (const params of directiveParams) {
              if (!directivePresence.has(params)) {
                const atRule = helpers.atRule({ name: 'tailwind', params })
                if (lastInserted) {
                  root.insertAfter(lastInserted, atRule)
                }
                else {
                  root.prepend(atRule)
                }
                lastInserted = atRule
                directivePresence.add(params)
              }
            }

            if (sourceInput?.file) {
              const tailwindcssConfig = (await resolveTailwindConfig(config, cwd, sourceInput)) ?? { content: [] }

              const basename = removeFileExtension(sourceInput.file)
              const contentEntries = [`${basename}.${extensionsGlob}`]

              const deps = await getDepFiles(`${basename}.wxml`)
              for (const dep of deps) {
                contentEntries.push(`${removeFileExtension(dep)}.${extensionsGlob}`)
              }

              const uniqueEntries = Array.from(new Set(contentEntries))
              const { content: existingContent } = tailwindcssConfig as { content?: Config['content'] }

              if (!existingContent || Array.isArray(existingContent)) {
                tailwindcssConfig.content = uniqueEntries
              }
              else if (typeof existingContent === 'object' && existingContent !== null) {
                const normalizedContent = { ...existingContent }
                const currentFiles = normalizedContent.files
                const normalizedFiles = Array.isArray(currentFiles)
                  ? currentFiles
                  : typeof currentFiles === 'string'
                    ? [currentFiles]
                    : []
                normalizedContent.files = Array.from(new Set([...normalizedFiles, ...uniqueEntries]))
                tailwindcssConfig.content = normalizedContent
              }
              else {
                tailwindcssConfig.content = uniqueEntries
              }

              await postcss([
                tailwindcss(tailwindcssConfig),
              ]).process(root, {
                from: sourceInput.file,
              }).async()
            }
          }
        },
      },
    ],
  }
}
creator.postcss = true

export default creator
