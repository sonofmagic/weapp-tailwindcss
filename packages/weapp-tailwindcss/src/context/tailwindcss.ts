import type { TailwindcssPatchOptions } from 'tailwindcss-patch'
import type { InternalUserDefinedOptions } from '@/types'
import { createRequire } from 'node:module'
import path from 'node:path'
import process from 'node:process'
import { fileURLToPath } from 'node:url'
import { logger } from '@weapp-tailwindcss/logger'
import { createTailwindcssPatcher } from '@/tailwindcss'
import { defuOverrideArray } from '@/utils'

const ENV_BASEDIR_KEYS = [
  'WEAPP_TAILWINDCSS_BASEDIR',
  'WEAPP_TAILWINDCSS_BASE_DIR',
  'TAILWINDCSS_BASEDIR',
  'TAILWINDCSS_BASE_DIR',
  'UNI_INPUT_DIR',
  'UNI_INPUT_ROOT',
  'UNI_CLI_ROOT',
  'UNI_APP_INPUT_DIR',
  'INIT_CWD',
  'PWD',
] as const

type EnvBasedirKey = typeof ENV_BASEDIR_KEYS[number]

const GENERIC_ENV_BASEDIR_KEYS = new Set<EnvBasedirKey>(['INIT_CWD', 'PWD'])

interface EnvBasedirResult {
  key: EnvBasedirKey
  value: string
}

function pickEnvBasedir(): EnvBasedirResult | undefined {
  for (const key of ENV_BASEDIR_KEYS) {
    const value = process.env[key]
    if (value && path.isAbsolute(value)) {
      return { key, value }
    }
  }
  return undefined
}

function pickPackageEnvBasedir(): string | undefined {
  const packageJsonPath = process.env.npm_package_json
  if (packageJsonPath) {
    const packageDir = path.dirname(packageJsonPath)
    if (packageDir && path.isAbsolute(packageDir)) {
      return packageDir
    }
  }
  const localPrefix = process.env.npm_config_local_prefix
  if (localPrefix && path.isAbsolute(localPrefix)) {
    return localPrefix
  }
  return undefined
}

function detectCallerBasedir(): string | undefined {
  const stack = new Error('resolveTailwindcssBasedir stack probe').stack
  if (!stack) {
    return undefined
  }
  if (process.env.WEAPP_TW_DEBUG_STACK === '1') {
    logger.debug('caller stack: %s', stack)
  }
  const lines = stack.split('\n')
  for (const line of lines) {
    const match = line.match(/\(([^)]+)\)/u) ?? line.match(/at\s+(\S.*)$/u)
    const location = match?.[1]
    if (!location) {
      continue
    }
    let filePath = location
    if (filePath.startsWith('file://')) {
      try {
        filePath = fileURLToPath(filePath)
      }
      catch {
        continue
      }
    }
    const [candidate] = filePath.split(':')
    const resolvedPath = path.isAbsolute(filePath) ? filePath : candidate
    if (!path.isAbsolute(resolvedPath)) {
      continue
    }
    if (resolvedPath.includes('node_modules') && resolvedPath.includes('weapp-tailwindcss')) {
      continue
    }
    try {
      return path.dirname(resolvedPath)
    }
    catch {
      continue
    }
  }
  return undefined
}

export function resolveTailwindcssBasedir(basedir?: string) {
  const envBasedirResult = pickEnvBasedir()
  const envBasedir = envBasedirResult?.value
  const envBasedirKey = envBasedirResult?.key
  const envBasedirIsGeneric = envBasedirKey ? GENERIC_ENV_BASEDIR_KEYS.has(envBasedirKey) : false
  const packageEnvBasedir = pickPackageEnvBasedir()
  const shouldDetectCaller = !envBasedir || envBasedirIsGeneric
  const callerBasedir = shouldDetectCaller ? detectCallerBasedir() : undefined
  const anchor = envBasedir ?? packageEnvBasedir ?? callerBasedir ?? process.cwd()
  if (process.env.WEAPP_TW_DEBUG_STACK === '1') {
    logger.debug('resolveTailwindcssBasedir anchor %O', {
      basedir,
      envBasedir,
      envBasedirKey,
      envBasedirIsGeneric,
      packageEnvBasedir,
      callerBasedir,
      npm_package_json: process.env.npm_package_json,
      cwd: process.cwd(),
      anchor,
    })
  }

  if (basedir && basedir.trim().length > 0) {
    if (path.isAbsolute(basedir)) {
      return path.normalize(basedir)
    }
    return path.resolve(anchor, basedir)
  }

  if (envBasedir && !envBasedirIsGeneric) {
    return path.normalize(envBasedir)
  }

  if (packageEnvBasedir) {
    return path.normalize(packageEnvBasedir)
  }

  if (callerBasedir) {
    const normalizedCaller = path.normalize(callerBasedir)
    const librarySegment = `${path.sep}weapp-tailwindcss${path.sep}`
    if (!normalizedCaller.includes(librarySegment)) {
      return normalizedCaller
    }
  }

  if (envBasedir) {
    return path.normalize(envBasedir)
  }

  const packageName = process.env.PNPM_PACKAGE_NAME
  if (packageName) {
    try {
      const anchorRequire = createRequire(path.join(anchor, '__resolve_tailwindcss_basedir__.cjs'))
      const packageJsonPath = anchorRequire.resolve(`${packageName}/package.json`)
      if (process.env.WEAPP_TW_DEBUG_STACK === '1') {
        logger.debug('package basedir resolved from PNPM_PACKAGE_NAME: %s', packageJsonPath)
      }
      return path.normalize(path.dirname(packageJsonPath))
    }
    catch {
      if (process.env.WEAPP_TW_DEBUG_STACK === '1') {
        logger.debug('failed to resolve package json for %s', packageName)
      }
      // ignore
    }
  }

  return path.normalize(process.cwd())
}

export function createTailwindcssPatcherFromContext(ctx: InternalUserDefinedOptions) {
  const {
    tailwindcssBasedir,
    supportCustomLengthUnitsPatch,
    tailwindcss,
    tailwindcssPatcherOptions,
    cssEntries,
    appType,
  } = ctx

  const resolvedTailwindcssBasedir = resolveTailwindcssBasedir(tailwindcssBasedir)
  ctx.tailwindcssBasedir = resolvedTailwindcssBasedir
  logger.debug('tailwindcss basedir resolved: %s', resolvedTailwindcssBasedir)

  type TailwindUserOptions = NonNullable<TailwindcssPatchOptions['tailwind']>

  const defaultTailwindcssConfig: TailwindUserOptions = {
    cwd: resolvedTailwindcssBasedir,
    v2: {
      cwd: resolvedTailwindcssBasedir,
    },
    v3: {
      cwd: resolvedTailwindcssBasedir,
    },
    v4: {
      base: resolvedTailwindcssBasedir,
      cssEntries,
    },
  }

  if (cssEntries?.length && (tailwindcss == null || tailwindcss.version == null)) {
    defaultTailwindcssConfig.version = 4
  }

  return createTailwindcssPatcher(
    {
      basedir: resolvedTailwindcssBasedir,
      cacheDir: appType === 'mpx' ? 'node_modules/tailwindcss-patch/.cache' : undefined,
      supportCustomLengthUnitsPatch: supportCustomLengthUnitsPatch ?? true,
      tailwindcss: defuOverrideArray<TailwindUserOptions, TailwindUserOptions[]>(
        (tailwindcss ?? {}) as TailwindUserOptions,
        defaultTailwindcssConfig,
      ),
      tailwindcssPatcherOptions,
    },
  )
}
