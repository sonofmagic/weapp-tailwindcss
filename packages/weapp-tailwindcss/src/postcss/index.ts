import postcss from 'postcss'
import type { IStyleHandlerOptions } from '../types'
import { defuOverrideArray } from '../utils'
import { getPlugins } from './plugins'

export function styleHandler(rawSource: string, options: IStyleHandlerOptions) {
  return postcss(getPlugins(options))
    .process(
      rawSource,
      options.postcssOptions?.options ?? {
        from: undefined,
      },
    )
    .async()
}

export function createStyleHandler(options: Partial<IStyleHandlerOptions>) {
  return (rawSource: string, opt?: Partial<IStyleHandlerOptions>) => {
    return styleHandler(
      rawSource,
      defuOverrideArray<IStyleHandlerOptions, Partial<IStyleHandlerOptions>[]>(opt as IStyleHandlerOptions, options),
    )
  }
}
