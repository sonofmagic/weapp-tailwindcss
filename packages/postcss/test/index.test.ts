import tailwindcss from '@tailwindcss/postcss'
import path from 'pathe'
import postcss from 'postcss'
import { createStyleHandler } from '@/index'

function generateCss(css: string, base: string) {
  return postcss([
    tailwindcss({
      base,
    }),
  ])
    .process(css, {
      from: './index.ts',
    })
}

describe('index', () => {
  it('only utilities', async () => {
    // https://developer.mozilla.org/en-US/docs/Web/CSS/calc-keyword
    const code = await generateCss('@import "weapp-tailwindcss";', path.resolve(__dirname, './fixtures/main'))
    expect(code.css).toMatchSnapshot()
    const styleHandler = createStyleHandler({
      isMainChunk: true,
    })
    const { css } = await styleHandler(code.css, {
      isMainChunk: true,
    })
    expect(css).toMatchSnapshot()
  })
})
