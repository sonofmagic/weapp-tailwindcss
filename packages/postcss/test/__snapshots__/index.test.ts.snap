// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`index > only utilities 1`] = `
"/*! tailwindcss v4.1.11 | MIT License | https://tailwindcss.com */
:root, :host {
  --color-red-500: oklch(63.7% .237 25.331);
  --color-blue-500: oklch(62.3% .214 259.815);
  --spacing: .25rem;
}
*,:after,:before,::backdrop {
  box-sizing: border-box;
  border: 0 solid;
  margin: 0;
  padding: 0;
}
.size-12 {
  width: calc(var(--spacing) * 12);
  height: calc(var(--spacing) * 12);
}
.rounded-full {
  border-radius: calc(infinity * 1px);
}
.rounded-se-full {
  border-start-end-radius: calc(infinity * 1px);
}
.bg-blue-500 {
  background-color: var(--color-blue-500);
}
.text-red-500\\/20 {
  color: color-mix(in srgb, oklch(63.7% .237 25.331) 20%, transparent);
  @supports (color: color-mix(in lab, red, red)) {
    color: color-mix(in oklab, var(--color-red-500) 20%, transparent);
  }
}"
`;

exports[`index > only utilities 2`] = `
"/*! tailwindcss v4.1.11 | MIT License | https://tailwindcss.com */
page,:host {
  --color-red-500: rgb(251, 44, 54);
  --color-blue-500: rgb(50, 128, 255);
  --spacing: .25rem;
}
@supports (color: color(display-p3 0 0 0%)) {
page,:host {
  --color-blue-500: rgb(50, 128, 255);
}

@media (color-gamut: p3) {
page,:host {
  --color-blue-500: color(display-p3 0.26642 0.49122 0.98862);
}
}
}
view,text,:after,:before,::backdrop {
  box-sizing: border-box;
  border: 0 solid;
  margin: 0;
  padding: 0;
}
.size-12 {
  width: calc(var(--spacing) * 12);
  height: calc(var(--spacing) * 12);
}
.rounded-full {
  border-radius: 9999px;
}
.rounded-se-full {
  border-top-right-radius: 9999px;
}
.bg-blue-500 {
  background-color: var(--color-blue-500);
}
.text-red-500s20 {
  color: rgba(251, 44, 54, 0.2);
}"
`;
